conan/tools/env/environment.py

```diff
@@ -51,22 +52,24 @@ def environment_wrap_command(env_filenames, env_folder, cmd, subsystem=None,
         raise ConanException("Cannot wrap command with different envs,"
                              "{} - {}".format(bats+ps1s, shs))
 
+    powershell = "powershell.exe" if conanfile.conf.get(
+        "tools.env.virtualenv:powershell") is True else conanfile.conf.get("tools.env.virtualenv:powershell")
+
```

Comment:
We have to check if this is True to provide compatibility with the deprecated syntax.
The problem is that `conanfile.conf.get("tools.env.virtualenv:powershell", check_type=str)` does not work because it will evaluate `True` or `False` as strings and not raise any exceptions.